### PR TITLE
Change git diff logic in custom skip integration script

### PR DESCRIPTION
**Pull request by furkatgofurov7**
This PR changes the way how we find out the changes made in the PR when custom skipping integration tests are used. The problem with the current implementation is, it does not consider the case, where for example the PR branch opened some time ago falls behind the origin branch it checked out (the branch is ahead of its checked out branch X commits and behind it, Y commits), then, git diff outputs all the changes that went in with Y commits since we are doing comparison from current branch against origin/{default_branch}. 
Now, this is done in a way that we look for all changes on the current branch since the branch start point as [here](https://stackoverflow.com/questions/29810331/is-there-a-quick-way-to-git-diff-from-the-point-or-branch-origin).

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]
/lgtm
Thanks for fixing! I was just looking at this and wondering why it wanted to run the tests for one of my PRs.

**Comment by furkatgofurov7** [comment]
> Thanks for fixing! I was just looking at this and wondering why it wanted to run the tests for one of my PRs.

This was caught thanks to your PR in CAPM3 :grinning:

**Review by smoshiur1237 — APPROVED** [review]
LGTM

**Review by namnx228 — APPROVED** [review]
LGTM

**Review by fmuyassarov — COMMENTED** [review]
/hold
@furkatgofurov7 can you please squash commits before merging
/approve

**Comment by furkatgofurov7** [comment]
> @furkatgofurov7 can you please squash commits before merging

I would rather prefer to keep it as is since I isolated them on purpose (1 to just change git diff logic and second for logging)

**Comment by fmuyassarov** [comment]
> > @furkatgofurov7 can you please squash commits before merging
> 
> I would rather prefer to keep it as is since I isolated them on purpose (1 to just change git diff logic and second for logging)

okay
/hold cancel

**Pull request merged**
